imports/server/garbage-collection.ts

```diff
@@ -7,38 +7,38 @@ import { Meteor } from "meteor/meteor";
 import { Random } from "meteor/random";
 import Logger from "../Logger";
 import Servers from "../lib/models/Servers";
+import ignoringDuplicateKeyErrors from "./ignoringDuplicateKeyErrors";
+import onExit from "./onExit";
 
 const serverId = Random.id();
 
 // Global registry of callbacks to run when we determine that a backend is dead.
-const globalGCHooks: ((deadServers: string[]) => void | Promise<void>)[] = [];
+const globalGCHooks: ((deadServers: string) => void | Promise<void>)[] = [];
 
 function registerPeriodicCleanupHook(
-  f: (deadServers: string[]) => void | Promise<void>,
+  f: (deadServer: string) => void | Promise<void>,
 ): void {
   globalGCHooks.push(f);
 }
 
-let firstUpsert = true;
 async function cleanup() {
-  const result = await Servers.upsertAsync(
-    { _id: serverId },
+  const updated = await Servers.updateAsync(
+    {
+      _id: serverId,
+      cleanupInProgressBy: undefined,
```

Comment:
I always forget: does this do the right thing, or does `undefined` get dropped from the selector entirely we need this selector to be `$exists: false`?

Reply:
Unfortunately I think this doesn't do what you think it does, at least in JS:

```
> await Models.Servers.find({ _id: "KH3KXvw4A98hQ74Ea", cleanupInProgressBy: undefined }).fetch()
[
  {
    _id: 'KH3KXvw4A98hQ74Ea',
    hostname: 'Mac.attlocal.net',
    pid: 51912,
    updatedAt: 2025-12-23T20:38:45.227Z,
    cleanupInProgressBy: 'KH3KXvw4A98hQ74Eb'
  }
]
```

I think the `undefined` gets dropped from the selector on the JS side before it gets to Mongo; you may be able to achieve what you were expecting with `null` but `undefined` here definitely is not excluding servers that have a `cleanupInProgressBy` set already.